src/main/java/by/itechart/api/service/impl/UserServiceImpl.java

```diff
@@ -35,6 +36,10 @@
 
     @Override
     public UserDTO create(CreateUserDTO createUserDTO) {
+        var fetchedUserWithThisEmail = userRepository.findByEmail(createUserDTO.getEmail());
+        fetchedUserWithThisEmail.ifPresent(user -> {
+            throw new UserEmailDuplicationException("User with this email is already exists");
```

Comment:
It would be more user-friendly to pass email into a message to point out which email is duplicated exactly